packages/@lwc/engine-dom/src/renderer.ts

```diff
@@ -143,211 +145,225 @@ if (isCustomElementRegistryAvailable()) {
-    },
-
-    isConnected(node: Node): boolean {
-        return node.isConnected;
-    },
-
-    insertGlobalStylesheet(content: string): void {
-        if (!isUndefined(globalStylesheets[content])) {
-            return;
+export function attachShadow(element: Element, options: ShadowRootInit): ShadowRoot {
+    if (hydrating) {
+        return element.shadowRoot!;
+    }
+    return element.attachShadow(options);
+}
+
+export function setText(node: Node, content: string): void {
+    node.nodeValue = content;
+}
+
+export function getProperty(node: Node, key: string): any {
+    return (node as any)[key];
+}
+
+export function setProperty(node: Node, key: string, value: any): void {
+    if (process.env.NODE_ENV !== 'production') {
+        if (node instanceof Element && !(key in node)) {
+            // TODO [#1297]: Move this validation to the compiler
+            assert.fail(
+                `Unknown public property "${key}" of element <${
+                    node.tagName
+                }>. This is likely a typo on the corresponding attribute "${htmlPropertyToAttribute(
+                    key
+                )}".`
+            );
         }
+    }
 
-        globalStylesheets[content] = true;
+    (node as any)[key] = value;
+}
 
-        const elm = document.createElement('style');
-        elm.type = 'text/css';
-        elm.textContent = content;
+export function getAttribute(
+    element: Element,
+    name: string,
+    namespace?: string | null
+): string | null {
+    return isUndefined(namespace)
+        ? element.getAttribute(name)
+        : element.getAttributeNS(namespace, name);
+}
 
-        globalStylesheetsParentElement.appendChild(elm);
-    },
+export function setAttribute(
+    element: Element,
+    name: string,
+    value: string,
+    namespace?: string | null
+): void {
+    return isUndefined(namespace)
+        ? element.setAttribute(name, value)
+        : element.setAttributeNS(namespace, name, value);
+}
 
-    insertStylesheet(content: string, target: Node): void {
-        if (supportsConstructableStyleSheets) {
-            insertConstructableStyleSheet(content, target);
-        } else {
-            // Fall back to <style> element
-            insertStyleElement(content, target);
-        }
-    },
+export function removeAttribute(element: Element, name: string, namespace?: string | null): void {
+    if (isUndefined(namespace)) {
+        element.removeAttribute(name);
+    } else {
+        element.removeAttributeNS(namespace, name);
+    }
+}
+
+export function addEventListener(
+    target: Node,
+    type: string,
+    callback: EventListener,
+    options?: AddEventListenerOptions | boolean
+): void {
+    target.addEventListener(type, callback, options);
+}
+
+export function removeEventListener(
+    target: Node,
+    type: string,
+    callback: EventListener,
+    options?: EventListenerOptions | boolean
+): void {
+    target.removeEventListener(type, callback, options);
+}
```

Comment:
The `addEventListener`/`removeEventListener` `options` were inconsistent before:

https://github.com/salesforce/lwc/blob/43bad75ca088bee45509aacbe230a6ce9df7bcd8/packages/%40lwc/engine-core/src/framework/renderer.ts#L29-L34

https://github.com/salesforce/lwc/blob/43bad75ca088bee45509aacbe230a6ce9df7bcd8/packages/%40lwc/engine-dom/src/renderer.ts#L239-L243